src/MoBi.Presentation/Tasks/Interaction/StartValuesTask.cs

```diff
@@ -305,5 +278,26 @@ public override IMoBiCommand GetAddCommand(TBuildingBlock itemToAdd, Module pare
       {
          return new AddBuildingBlockToModuleCommand<TBuildingBlock>(itemToAdd, parent);
       }
+
+      protected (MoBiSpatialStructure spatialStructure, MoleculeBuildingBlock moleculeBuildingBlock) SelectBuildingBlocksForExtend(bool moleculeRequired = true)
+      {
+         var moleculeBlockCollection = _interactionTaskContext.BuildingBlockRepository.MoleculeBlockCollection;
+         var spatialStructureCollection = _interactionTaskContext.BuildingBlockRepository.SpatialStructureCollection;
+         if ((!moleculeRequired || moleculeBlockCollection.Count == 1) && spatialStructureCollection.Count == 1)
+         {
+            return (spatialStructureCollection.Single(), moleculeRequired ? moleculeBlockCollection.Single() : null);
+         }
+
+         using (var selectorPresenter = Context.Resolve<ISelectBuildingBlocksForExtendPresenter>())
+         {
+            selectorPresenter.SelectBuildingBlocksForExtend(moleculeRequired);
+            if (selectorPresenter.SelectedSpatialStructure == null || (moleculeRequired && selectorPresenter.SelectedMoleculeBuildingBlock == null))
+            {
+               return (null, null);
+            }
+
+            return (selectorPresenter.SelectedSpatialStructure, selectorPresenter.SelectedMoleculeBuildingBlock);
```

Comment:
That was in case a user selects one but not  the other. But that actually can't happen with the validation anyway. On view cancel, the selections are cleared.